src/helpers/queries.js

```diff
@@ -0,0 +1,21 @@
+import { Q } from 'cozy-client'
+import { DOCTYPE_CONTACTS } from './doctypes'
+
+export const queryContactById = {
+  definition: id => () => Q(DOCTYPE_CONTACTS).getById(id),
+  options: {
+    as: 'contactById'
+  }
+}
+export const queryAllGroups = {
+  definition: () =>
+    Q('io.cozy.contacts.groups')
+      .where({
+        trashed: { $exists: false }
+      })
+      .sortBy([{ name: 'asc' }])
+      .indexFields(['name']),
```

Comment:
1. Why do we need the `indexFields` here ? I thought that cozy-client would automatically index the fields in the sortBy ?

2. Here, if a group has { trashed: false }, it will not be retrieved, is it expected ?

Reply:
Adding a fetchPolicy would be good here.

Reply:
1. We don't need `indexFields`, I know now that it's not necessary 👍 
2. Yes it is

Reply:
2. It seems counter-intuitive. Is this design decision discussed somewhere ?

Reply:
https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.contacts.md => `trashed?: {boolean} is true if the group is marked for removal and will be deleted soon` is clear enough? 

But even with that, I'm wondering why we ask this question? We want to display the `groups` the user have created in order to be able to set them on a contact or to display a list to filter the contacts' list. It makes sense to not fetch the deleted ones, no?

Reply:
> Adding a fetchPolicy would be good here.

I tried but the fetch status of the request stay still in 'loading' during the time of the fetch policy. Expl: with the fetchPolicy at 5sec, the fetch status is loaded at T0, remain in 'loading' during T1 to T5, then works again at T5+ (if we trigger the request at T0, between T1 and T5, and after T5).

I looked at https://docs.cozy.io/en/cozy-client/api/cozy-client/#fetchpolicies and https://docs.cozy.io/en/cozy-client/react-integration/, get some examples from Banks and Drive to find some clues, but I think there is a problem with the store maybe ?

Reply:
Yep, the fetchStatus will still be loading, but now : 
- you have a lastFetch attribute in the query result https://github.com/cozy/cozy-banks/blob/master/src/ducks/client/utils.js#L16-L18
- https://github.com/cozy/cozy-banks/blob/40338ac6133c8b946ea60f6c3aece3dbc264cf1a/src/ducks/brandDictionary/selectors.js#L10-L12

Reply:
> It makes sense to not fetch the deleted ones, no?

My question is for the groups with { trashed: false }, not the ones with { trashed: true } ;)

Reply:
Oh ok I saw and tested this lastFetch but didn't ring any bells :( Probably didn't understand how to use it. Thanks for the tips

Reply:
FYI, the fetchStatus still in loading is a bug but yes you can circumvent this with lastFetch.

Reply:
> My question is for the groups with { trashed: false }, not the ones with { trashed: true } ;)

Holy sh** 🤕. My bad. 

In the Contacts App, we can not have a `trashed: false` but you right, we can be more flexible with a better selector: 
```
"$or": [
         {
            "trashed": {
               "$exists": false
            }
         },
         {
            "trashed": false
         }
      ]
```